packages/material-ui/src/TextField/TextField.js

```diff
@@ -158,12 +159,19 @@ const TextField = React.forwardRef(function TextField(props, ref) {
       {...other}
     >
       {label && (
-        <InputLabel htmlFor={id} ref={labelRef} {...InputLabelProps}>
+        <InputLabel htmlFor={id} ref={labelRef} id={inputLabelId} {...InputLabelProps}>
           {label}
         </InputLabel>
       )}
       {select ? (
-        <Select aria-describedby={helperTextId} value={value} input={InputElement} {...SelectProps}>
+        <Select
+          aria-describedby={helperTextId}
+          id={id}
```

Comment:
Is this needed? I expect it to already come from the `InputElement` element.

Reply:
I'll add a test/try without it. I think I considered this but it might have been in another context.

Reply:
It is required. But it helped uncover another bug where the input[type="hidden"] had a (colliding) id and `aria-describedby`. Neither of which are required anymore.